### PR TITLE
docs: document controller runtime, resources and talosctl get

### PR DESCRIPTION
**Pull request by smira**
This is more of a in-depth guide explaining internals.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by Unix4ever — DISMISSED** [review]
Looks good to me from the contents perspective :) 
Probably also makes sense to update all resources return values as they are a bit different after the naming update PR.

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    # Provides machine specific configuration options.
+...
+```
+
+Getting control plane static pod statuses:
+
+```bash
+$ talosctl get staticpodstatus
+NODE         NAMESPACE      TYPE              ID                                                           VERSION   READY
+172.20.0.2   controlplane   StaticPodStatus   kube-system/kube-apiserver-talos-default-master-1            3         True
```
[the remark this answers is not in the document]
Comment (smira) [reply website/content/docs/v0.9/Learn More/controllers-resources.md:178]: example here should be accurate, full type name is `StaticPodStatuses.kubernetes.talos.dev`, but we show `DisplayType` which is `StaticPodStatus` (similar to Kubernetes, which doesn't use full type name in table output).

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+title: "Controllers and Resources"
+weight: 8
+---
+
+<!-- markdownlint-disable MD038 -->
+
+Talos implements concepts of *resources* and *controllers* to implement internal operations of the
+operating system. Talos resources are controllers are very similar to Kubernetes resources and controllers,
+but there are some differences. Content of this document is not required to operate Talos, but it is
+useful for troubleshooting Talos.
```
[the remark this answers is not in the document]
Comment (smira) [reply website/content/docs/v0.9/Learn More/controllers-resources.md:11]: I wonder if I should just drop `Talos` there, it should be clear without it.

File: website/content/docs/v0.9/Learn More/controllers-resources.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Unix4ever) [reply website/content/docs/v0.9/Learn More/controllers-resources.md:178]: yeah, sorry. I just had an old version of talsoctl :facepalm:

File: website/content/docs/v0.9/Learn More/controllers-resources.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewrynhard) [reply website/content/docs/v0.9/Learn More/controllers-resources.md:11]: Should be clear without it.

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
+---
+title: "Controllers and Resources"
+weight: 8
+---
+
+<!-- markdownlint-disable MD038 -->
+
+Talos implements concepts of *resources* and *controllers* to implement internal operations of the
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:8]:
```suggestion
Talos implements concepts of *resources* and *controllers* to facilitate internal operations of the
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+operating system. Talos resources are controllers are very similar to Kubernetes resources and controllers,
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:9]:
```suggestion
operating system. Talos resources and controllers are very similar to Kubernetes resources and controllers,
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+but there are some differences. Content of this document is not required to operate Talos, but it is
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:10]:
```suggestion
but there are some differences. The content of this document is not required to operate Talos, but it is
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+useful for troubleshooting Talos.
+
+Starting with Talos 0.9, most of the Kubernetes control plane boostrapping and operations is implemented
+via controllers and resources which allows Talos to be reactive to configuration changes, environment
+changes (e.g. time sync).
+
+## Resources
+
+Resource captures piece of system state. Each resource belongs to some type which defines resource contents.
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:19]:
```suggestion
A resource captures a piece of system state. Each resource belongs to a "Type" which defines resource contents.
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+duplicate resource IDs.
+
+At the moment of this writing, all resources are local to the node and stored in memory. So on every reboot
+resource state is rebuilt from scratch (the only exception is `MachineConfig` resource which reflects current
+machine config).
+
+## Controllers
+
+Controllers are running as independent lightweight threads in Talos. Goal of the controller is to reconcile
+the state based on the inputs and update outputs.
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:35]:
```suggestion
Controllers run as independent lightweight threads in Talos. The goal of the controller is to reconcile
the state based on inputs and eventually update outputs.
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+
+Controller can have any number of resource types (and namespaces) as input, in other words it watches
+specified resources for changes and reconciles on input resource change. Controller might have additional
+inputs: running reconcile on schedule, watching `etcd` keys, etc.
+
+Controller has single output: set of resources of fixed type in a fixed namespace. Only one controller
+can manage resource type in the namespace, so conflicts are avoided.
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:42]:
```suggestion
A controller has a single output: a set of resources of fixed type in a fixed namespace. Only one controller
can manage resource type in the namespace, so conflicts are avoided.
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+NODE         NAMESPACE   TYPE        ID             VERSION
+172.20.0.2   meta        Namespace   config         1
+```
+
+### Output
+
+Command `talosctl get` supports following output modes:
+
+* `table` (default) prints resource list as a table
+* `yaml` prints pretty formatted resources with details, including full metadata spec, this format carries most
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:122]:
```suggestion
* `yaml` prints pretty formatted resources with details, including full metadata spec. This format carries most
```

File: website/content/docs/v0.9/Learn More/controllers-resources.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+  details from the backend resource (e.g. comments in `MachineConfig` resource)
+* `json` prints same information as `yaml`, some additional details (e.g. comments) might be lost,
+  this format is useful for automated processing with tools like `jq`
+
+### Watching Changes
+
+If flag `--watch` is appended to the `talosctl get` command, command switches to watch mode. If list of resources
```
Comment (rsmitty) [inline website/content/docs/v0.9/Learn More/controllers-resources.md:129]:
```suggestion
If flag `--watch` is appended to the `talosctl get` command, the command switches to watch mode. If list of resources
```

**Review by rsmitty — APPROVED** [review]

**Pull request merged**